src/model/channel/channel_id.rs

```diff
@@ -414,7 +414,16 @@ impl ChannelId {
             }
         }
 
-        cache_http.http().get_channel(self.0).await
+        let channel = cache_http.http().get_channel(self.0).await?;
+        #[cfg(feature = "cache")]
+        {
+            if let Some(cache) = cache_http.cache() {
+                if let Some(guild_channel) = channel.clone().guild() {
```

Comment:
Could replace this with `if let Channel::Guild(guild_channel) = &channel` and clone only if the if succeeds, to get rid of a useless clone() on non-guild channels. Although performance-wise it won't matter